feat: update to CaDiCaL version 3.0.0

Dekker1 opened the pull request.

Dekker1 committed:
feat: update to CaDiCaL version 2.2.0

AllenZzw commented:
Fix `int_pow` in https://github.com/AllenZzw/huub/commit/5ad053af611843b05a67701ee1c5263e3dac346f

Dekker1 commented:
Thank you for the `IntPowBounds` fix @AllenZzw. Did you manage to find where (and if) we can switch off the new presolving functionality?

AllenZzw commented:
I think we can switch off the new lightweight preprocessing by setting the `preprocesslight` option to 0. (https://github.com/arminbiere/cadical/blob/7b99c07f0bcab5824a5a3ce62c7066554017f641/src/options.hpp#L162)

AllenZzw commented:
I also did a round of test on MiniZinc challenge instances from 2023 to 2025. There are some failures which relate to `int_value_procede`, `cumulative` and `BoolFormula`. I will have a look at this.

AllenZzw commented:
> Thank you for the `IntPowBounds` fix @AllenZzw. Did you manage to find where (and if) we can switch off the new presolving functionality?

Light preprocessing is disabled in https://github.com/AllenZzw/huub/commit/0b4b76db182ba7d2a6f31d9ee1d66776c851f21e

Dekker1 committed:
feat: add preprocessing-light option for CaDiCaL oracle

Dekker1 committed:
refactor: use `--cadical-` prefix for CaDiCaL specific options

This also add the following options which are important for performance
when using CaDiCaL 3.0.0:
- preprocessing-light
- reduce-type
- reduce-interval

Dekker1 commented:
I've added it as a separate CLI option. It is a shame that it does not help. In general, it appears that the new version isn't necessarily beneficial.

I think it might be worthwhile to merge this, but mostly just to stay up-to-date with the CaDiCaL development. 

Do you have any further concerns before we merge @AllenZzw?

AllenZzw commented:
It appears that there are some fixs in 2.2.1 https://github.com/arminbiere/cadical/tree/rel-2.2.1. Do we want to merge that? 

CaDiCal 3.0.0 is also released several weeks ago, but there are some API changes. Maybe we would like to merge this version later.

Dekker1 committed:
feat: update to CaDiCaL version 3.0.0
